api/ombpdf/footnotes.py

```diff
@@ -107,6 +107,8 @@ def finish_footnote():
                 if match:
                     finish_footnote()
                     footnote, desc = match.groups()
+                    for char in line[:len(footnote) + 1]:
```

Comment:
This line threw me for a bit:

I initially thought interpreted "footnote" to mean the whole line, which would make the length redundant. Perhaps we should swap the variable name to something like "number" or "marker" to be explicit ("marker" may be more consistent with the rest of the script)? Unfortunately "footnote" can mean different concepts at different times.

Following that, I didn't understand where "+ 1" came from. I *think* it's ensuring the space is grabbed. But why would we want the space to be tagged as a footnote marker?

Reply:
Good points, I just refactored the code in 4d2ad60d5a42f9bbbf7530387d802a1c55eba273 to make this more readable.

The `+ 1` was indeed the space following the footnote; it was a bit of a toss-up as to whether to classify it as part of the marker, and I decided to do it so that it wasn't mistaken to be part of the actual footnote text by downstream clients.  That said, I've moved it into the captured group of the marker in the regexp so that it's not a confusing `+ 1` anymore, at least.